src/engine/readaheadmanager.cpp

```diff
@@ -175,21 +175,23 @@ void ReadAheadManager::hintReader(double dRate, HintVector* pHintList) {
 
     // SoundTouch can read up to 2 chunks ahead. Always keep 2 chunks ahead in
     // cache.
-    SINT length_to_cache = 2 * CachingReaderChunk::kSamples;
+    SINT frameCountToCache = 2 * CachingReaderChunk::kFrames;
 
     // this called after the precious chunk was consumed
-    int sample = SampleUtil::roundPlayPosToFrameStart(
-            m_currentPosition, kNumChannels);
-    current_position.length = length_to_cache;
-    current_position.sample = in_reverse ?
-            sample - length_to_cache :
-            sample;
+    SINT frame = SampleUtil::roundPlayPosToFrameStart(
```

Comment:
It feels and looks a bit odd to go from sample to frames to samples and finally back to frames